### ISSUE TITLE
TypeError: The input to be decoded is not correctly encoded - when upgrading from 0.7.6 to 0.8.6

### ISSUE THREAD

**Issue by MartinXPN** [awinogrodzki/next-firebase-auth-edge#92]
I was trying to upgrade the library version from `0.7.6` to the latest one `0.8.6` and none of the currently logged-in users could use the app. 

For all users logged in with the previous version (0.7.6), the app started to refresh every second, not letting them interact with it.
Looking at the logs on the server, I see the following:

```
[GET] [middleware: "middleware"] /api/login reason=EDGE_FUNCTION_INVOCATION_FAILED, status=500, user_error=true
```
and
```
TypeError: The input to be decoded is not correctly encoded.
    at (node_modules/jose/dist/browser/runtime/base64url.js:35:14)
    at (node_modules/next-firebase-auth-edge/lib/auth/rotating-credential.js:12:35)
    at (node_modules/next-firebase-auth-edge/lib/auth/rotating-credential.js:16:20)
    at (node_modules/next-firebase-auth-edge/lib/auth/cookies/sign.js:11:38)
    at (node_modules/next-firebase-auth-edge/lib/next/cookies.js:34:98)
    at (node_modules/next-firebase-auth-edge/lib/next/cookies.js:59:11)
    at (node_modules/next/dist/esm/server/web/adapter.js:145:19)
```

Seems like the app is not able to properly get or set the user from the cookies.
Have I done something wrong? Is there any way to prevent this from happening?
Thanks a lot for the help!

**Comment by MartinXPN** [comment]
After making several changes and running the app locally (both prod build and dev build), I got the following errors:

```
- error node_modules/next-firebase-auth-edge/lib/auth/token-verifier.js (80:15) @ FirebaseTokenVerifier.mapJoseErrorToAuthError
- error Internal error: fetch failed
```


```
Error: Internal error: fetch failed
at FirebaseTokenVerifier.mapJoseErrorToAuthError (webpack-internal:///(middleware)/./node_modules/next-firebase-auth-edge/lib/auth/token-verifier.js:79:16)
at eval (webpack-internal:///(middleware)/./node_modules/next-firebase-auth-edge/lib/auth/token-verifier.js:65:24)
at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
at async FirebaseTokenVerifier.decodeAndVerify (webpack-internal:///(middleware)/./node_modules/next-firebase-auth-edge/lib/auth/token-verifier.js:33:9)

at async FirebaseTokenVerifier.verifyJWT (webpack-internal:///(middleware)/./node_modules/next-firebase-auth-edge/lib/auth/token-verifier.js:21:25)
at async verifyIdToken (webpack-internal:///(middleware)/./node_modules/next-firebase-auth-edge/lib/auth/index.js:130:32)
at async getCustomIdAndRefreshTokens (webpack-internal:///(middleware)/./node_modules/next-firebase-auth-edge/lib/auth/index.js:156:24)
at async setAuthCookies (webpack-internal:///(middleware)/./node_modules/next-firebase-auth-edge/lib/next/cookies.js:54:32)
at async adapter (webpack-internal:///(middleware)/./node_modules/next/dist/esm/server/web/adapter.js:161:20)
at async runWithTaggedErrors (/node_modules/next/dist/server/web/sandbox/sandbox.js:98:24)
```

**Comment by awinogrodzki** [comment]
Hey @MartinXPN, thanks for reporting :-)

Could you run `npx next info` in your project and paste the result here? It should give me some context on Node and Next versions you're using. Also, could you share a list of `firebase` sdk packages your app depends on?

**Comment by MartinXPN** [comment]
Sure, here is the result of `npx next info`:
```
    Operating System:
      Platform: darwin
      Arch: x64
      Version: Darwin Kernel Version 21.6.0: Thu Jul  6 22:18:26 PDT 2023; root:xnu-8020.240.18.702.13~1/RELEASE_X86_64
    Binaries:
      Node: 18.17.0
      npm: 9.6.7
      Yarn: 1.22.19
      pnpm: N/A
    Relevant Packages:
      next: 13.4.19
      eslint-config-next: 13.4.19
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.2.2
    Next.js Config:
      output: N/A
```

I'm currently using several firebase-related packages:
```
    "firebase": "^10.4.0",
    "firebase-admin": "^11.10.1",
    "firebaseui": "^6.1.0",
    "next-firebase-auth-edge": "^0.8.6",
```

**Comment by awinogrodzki** [comment]
I cannot reproduce the issue with those specific versions on https://next-firebase-auth-edge-starter.vercel.app/, but I have a feeling it's connected with the value of `cookieSignatureKeys`. Not 100% sure though.

Could you install `next-firebase-auth-edge@0.8.7-1` and check if the error is still happening?

**Comment by awinogrodzki** [comment]
If it does still happen, will it start working after clearing browser cache?

**Comment by MartinXPN** [comment]
Yeah, I just tried the `next-firebase-auth-edge@0.8.7-1` version and it worked! 
Thank you so much!
May I ask what was the issue?

**Comment by awinogrodzki** [comment]
Here's the PR with the fix https://github.com/awinogrodzki/next-firebase-auth-edge/pull/93/files

Before `0.8.7-1`, I was using `base64url.decode` from `jose`  to convert cookie signature key to `UInt8Array` to generate cookie signature.

I think `cookieSignatureKeys` you pass to `authentication` middleware function have some unusual character or set of characters that cannot be decoded by `base64url.decode` function. It was mistake on my side. I tested `base64url.decode` against number of combinations and assumed it was safe for any string, but the assumption was false.

Anyways, converting to `UInt8Array` directly is much more efficient in this case. I'm glad the fix have worked!

**Comment by MartinXPN** [comment]
Yeah, works like a charm. Thank you so much for fixing it this quickly!

**Comment by awinogrodzki** [comment]
Fix released in `0.8.7`
